src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayToFerryTagCheck.java

```diff
@@ -0,0 +1,134 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.openstreetmap.atlas.checks.atlas.predicates.TypePredicates;
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.FerryTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.RouteTag;
+import org.openstreetmap.atlas.tags.Taggable;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Based on the osm wiki, when route=ferry exists, there should be a ferry=* tag where the values of
+ * ferry are highway values. This check flags all edges that have both {@link HighwayTag} and
+ * {@link RouteTag} value equal to "FERRY".
+ *
+ * @author sayas01
+ */
+public class HighwayToFerryTagCheck extends BaseCheck
+{
+    private static final String MINIMUM_HIGHWAY_TYPE_DEFAULT = HighwayTag.PATH.toString();
+    private static final String FERRY_TAG_IF_SAME_AS_HIGHWAY_INSTRUCTION = "This way {0,number,#} has a Ferry and a Highway tag for a ferry route. "
+            + "Please verify and remove the highway tag.";
+    private static final String FERRY_TAG_IF_ABSENT_INSTRUCTION = "This way {0,number,#} has a Highway tag for a ferry route instead of a Ferry tag. "
+            + "Please verify and add a Ferry tag with the Highway tag value and remove the highway tag.";
+    private static final String FERRY_TAG_IF_DIFFERENT_FROM_HIGHWAY_INSTRUCTION = "This way {0,number,#} has a Ferry and a Highway tag for a ferry route. "
+            + "Please verify and update the Ferry tag with the Highway tag value and remove the highway tag.";
+    private static final Predicate<Taggable> HAS_FERRY_TAG = object -> Validators
+            .hasValuesFor(object, FerryTag.class);
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            FERRY_TAG_IF_DIFFERENT_FROM_HIGHWAY_INSTRUCTION,
+            FERRY_TAG_IF_SAME_AS_HIGHWAY_INSTRUCTION, FERRY_TAG_IF_ABSENT_INSTRUCTION);
+
+    private final HighwayTag minimumHighwayType;
+
+    /**
+     * Default constructor
+     *
+     * @param configuration
+     *            {@link Configuration} required to construct any Check
+     */
+    public HighwayToFerryTagCheck(final Configuration configuration)
+    {
+        super(configuration);
+        final String highwayType = (String) this.configurationValue(configuration,
+                "highway.type.minimum", MINIMUM_HIGHWAY_TYPE_DEFAULT);
+        this.minimumHighwayType = Enum.valueOf(HighwayTag.class, highwayType.toUpperCase());
```

Comment:
Could you use the other `configurationValue` function above and put this line (converting from string to highway tag) as a lambda as the transform parameter?